apps/studio/src/components/features/editing-experience/PageCreateModal.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useEffect, useState } from 'react'
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputLeftAddon,
+  Text,
+  Stack,
+  Box,
+} from '@chakra-ui/react'
+import {
+  Button,
+  FormErrorMessage,
+  ModalCloseButton,
+} from '@opengovsg/design-system-react'
+import { useForm, SubmitHandler } from 'react-hook-form'
+interface PageCreateModalProps {
+  isOpen: boolean
+  onClose: () => void
+}
+
+type PageCreateFormFields = {
+  title: string
+  url: string
+}
+
+/* TODO: Can extract these out to a constants file if can be reused */
+const ERROR_MESSAGES = {
+  MIN_LENGTH: 'Minimum length should be 1',
+  MAX_LENGTH: (max: number) => `Maximum length should be ${max}`,
+  TITLE: {
+    REQUIRED: 'Enter a title for this page',
+  },
+  URL: {
+    REQUIRED: 'Enter a URL for this page.',
+  },
+}
+
+export const PageCreateModal = ({
+  isOpen,
+  onClose,
+}: PageCreateModalProps): JSX.Element => {
+  const MAX_TITLE_LENGTH = 100
+  const MAX_PAGE_URL_LENGTH = 150
+  const [titleLen, setTitleLen] = useState(0)
+  const [pageUrlLen, setPageUrlLen] = useState(0)
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<PageCreateFormFields>()
+
+  const watchAllFields = watch()
+  console.log(watchAllFields)
+  console.log(`errors`, errors)
+  useEffect(() => {
+    setTitleLen(watchAllFields.title?.length || 0)
+    setPageUrlLen(watchAllFields.url?.length || 0)
+  }, [watchAllFields])
+
+  /* TODO: When integrating with BE */
+  const onSubmit: SubmitHandler<PageCreateFormFields> = (data) =>
+    console.log(data)
```

Comment:
[FYI, no action required] the dto's PR on my end might have the trpc call ready -> just need to call it when merged

Reply:
gotcha